### ISSUE TITLE
AccessibilityInfo.setAccessibilityFocus does not work with Samsung Voice Assistant

### ISSUE THREAD

**Issue by Shanghai-** [facebook/react-native#28674]
## Description
According to the documentation, [AccessibilityInfo.setAccessibilityFocus](https://reactnative.dev/docs/accessibilityinfo#setaccessibilityfocus) should set the screen reader's focus to the given component across all platforms. However, on Samsung devices the call does nothing; that is, focus remains where it was before setAccessibilityFocus was called. The function works as expected in iOS and on Google phones. 

The accessibility documentation contains an [Android-specific example of setting focus](https://reactnative.dev/docs/accessibility#sending-accessibility-events-android) using the native Accessibility Event system, but that code does not work on Samsung phones either. According to the AccessibilityInfo docs, setAccessibilityFocus uses the AccessibilityEvent system behind the scenes, so it makes sense that it wouldn't work to use it directly either. 

**This is a major accessibility issue,** as it makes building accessible overlays, dropdown menus, etc. extremely difficult. A fix would be very useful for the community.

## React Native version:
```
System:
    OS: macOS 10.15.3
    CPU: (8) x64 Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz
    Memory: 84.73 MB / 16.00 GB
    Shell: 5.7.1 - /bin/zsh
  Binaries:
    Node: 12.13.0 - /usr/local/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.13.4 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.8.4 - /usr/local/bin/pod
  SDKs:
    iOS SDK: Not Found
    Android SDK: Not Found
  IDEs:
    Android Studio: 3.5 AI-191.8026.42.35.5900203
    Xcode: /undefined - /usr/bin/xcodebuild
  Languages:
    Java: 13.0.1 - /usr/bin/javac
    Python: 2.7.16 - /usr/bin/python
  npmPackages:
    @react-native-community/cli: Not Found
    react: ^16.13.1 => 16.13.1 
    react-native: 0.62.2 => 0.62.2 
  npmGlobalPackages:
    *react-native*: Not Found
```

## Steps To Reproduce
1. Open the attached Expo snack on a physical Samsung device
2. Enable Voice Assistant (Settings > Accessibility > Screen Reader > Voice Assistant)
3. Navigate back to the Expo app and highlight the button labeled "Select to set focus to bottom text"
4. Double tap to activate the button and observe.

## Expected Results
Focus should shift to the text at the bottom of the screen, "Jump to me!".

## Snack, code example, screenshot, or link to a repository:
The repro can be found in this snack: https://snack.expo.io/mMl5DkO!7
I was also able to reproduce the issue in a standalone Typescript project: https://github.com/Shanghai-/ReactNativeFocusProblem

**Comment by babylone-star** [comment]
+1. Can confirm same problem on a Samsung Galaxy Note S9

**Comment by jessamp** [comment]
Hi! I can also repro this on my tab s6 and galaxy s20. Wondering if there's plans to add this feature moving forward?

**Comment by JordanDuncan** [comment]
Can confirm repro on Galaxy S9 running Android 10.

**Comment by elfstyle** [comment]
Doesn't work on S10.

**Comment by nico22utn** [comment]
Hi! Just one question, this issue generates a problem reading the `accessibilityHeader='header'` too?

**Comment by amarlette** [comment]
@blavalla has not been able to reproduce this issue on the current RN version. Was this issue resolved by updates since the issue was created?

**Comment by blavalla** [comment]
Just as a note, if you can reproduce this still, could you let me know the version of Android, and the version of Voice Assistant you are using? Thanks!

Note: In the latest Samsung devices, "Voice Assistant" has been renamed "Talkback".  It was always actually a fork of Google's Talkback, just an often out-of-date one. The newest version is up to date as of earlier this year however, and fixes many issues that older versions had. It may not be available on all devices however, as Samsung bundles it as part of its OneUI skin.  So if we *can* fix an issue in an older version of Voice Assistant, we should, but we may not need to if the version affected here is no longer in the wild.

Thanks!

**Comment by blavalla** [comment]
Also just to check, if you *can* reproduce the issue still, could you try installing Google's Talkback (part of the Android Accessibility Suite) from the Play Store?  If the issue does reproduce in Voice Assistant, and does not reproduce in Talkback, then it's likely a bug on that side of things, and there isn't much we can do.

**Comment by blavalla** [comment]
Feel free to re-open if this comes up again, but at this point Samsung has abandoned Voice Assistant and now ships devices with Talkback.  Closing out.
